packages/rainbowkit-siwe-next-auth/src/useSiweNextAuth.ts

```diff
@@ -0,0 +1,57 @@
+import {
+  AuthenticationConfig,
+  createAuthenticationAdapter,
+} from '@rainbow-me/rainbowkit';
+import { getCsrfToken, signIn, signOut, useSession } from 'next-auth/react';
+import { useMemo } from 'react';
+import { SiweMessage } from 'siwe';
+
+export interface UseSiweNextAuthOptions {
+  statement?: string;
+}
+
+export function useSiweNextAuth({
+  statement = 'Sign in with Ethereum to the app.',
+}: UseSiweNextAuthOptions = {}): AuthenticationConfig<SiweMessage> {
+  const { status } = useSession();
+  const adapter = useMemo(
+    () =>
+      createAuthenticationAdapter({
+        createMessage: ({ address, chainId, nonce }) =>
+          new SiweMessage({
```

Comment:
does the consumer need to pass any of these too?

https://github.com/spruceid/siwe/blob/main/packages/siwe/lib/client.ts#L41-L54

Reply:
Some of those values are too dynamic to be passed in via the provider. I wonder if this means we need to support passing in a function to dynamically resolve them.

e.g.

```tsx
function createMessage({ address, chainId, nonce }) {
  return new SiweMessage({
    address,
    chainId,
    nonce,
    ...etc
   });
}

<RainbowKitSiweNextAuthProvider createMessage={createMessage}>
```

Note that this means `siwe` would need to be a peer dep.

Or maybe just providing the options?

```tsx
function getMessageOptions({ address, chainId, nonce }) {
  return ({
    address,
    chainId,
    nonce,
    requestId: '...'
  });
}

<RainbowKitSiweNextAuthProvider getMessageOptions={getMessageOptions}>
```

This could even be simplified to provide defaults?

```tsx
function getMessageOptions({ defaults }) {
  return ({
    ...defaults,
    requestId: '...'
  });
}

<RainbowKitSiweNextAuthProvider getMessageOptions={getMessageOptions}>
```

Or further simplified so that we always merge their options with the defaults.

```tsx
function getMessageOptions() {
  return ({
    requestId: '...'
  });
}

<RainbowKitSiweNextAuthProvider getMessageOptions={getMessageOptions}>
```

The last one might be a step too far, but it's interesting. What do you think?

Reply:
I've added a new prop called `getSiweMessageOptions` that follows the last example in my previous comment, plus a type called `GetSiweMessageOptions` to make it easier to define this function statically. I've also updated the docs to reflect this. Feeling pretty good about how flexible this is.